packages/cli/src/commands/proxy/index.ts

```diff
@@ -0,0 +1,87 @@
+import * as http from 'node:http';
+
+import { isValidUrl } from '@logto/core-kit';
+import chalk from 'chalk';
+import type { CommandModule } from 'yargs';
+
+import { consoleLog } from '../../utils.js';
+
+import { type ProxyCommandArgs } from './types.js';
+import { createLogtoResponseHandler, createProxy, isLogtoRequestPath } from './utils.js';
+
+const proxy: CommandModule<unknown, ProxyCommandArgs> = {
+  command: ['proxy'],
+  describe: 'Command for Logto proxy',
+  builder: (yargs) =>
+    yargs
+      .options({
+        'experience-uri': {
+          alias: ['x'],
+          describe: 'The URI of your custom sign-in experience page.',
+          type: 'string',
+        },
+        'tenant-id': {
+          alias: ['t'],
+          describe:
+            'Your Logto Cloud tenant ID. WHen provided, endpoint URI will be set to `https://<tenant-id>.logto.app` by default.',
+          type: 'string',
+        },
+        endpoint: {
+          alias: 'ep',
+          describe:
+            'Specify the full Logto endpoint URI, which takes precedence over tenant ID when provided.',
+          type: 'string',
+        },
+        port: {
+          alias: 'p',
+          describe: 'The port number where the proxy server will be running on. Defaults to 9000.',
+          type: 'number',
+          default: 9000,
+        },
+      })
+      .global('e'),
+  handler: async ({ 'experience-uri': expUri, 'tenant-id': tenantId, endpoint, port }) => {
+    if (!expUri || !isValidUrl(expUri)) {
+      consoleLog.fatal(
+        'A valid sign-in experience URI must be provided. E.g.: http://localhost:4000'
+      );
+    }
+    if (!tenantId && (!endpoint || !isValidUrl(endpoint))) {
+      consoleLog.fatal('Either tenant ID or a valid Logto endpoint URI must be provided.');
+    }
+
+    const logtoEndpointUrl = new URL(endpoint ?? `https://${tenantId}.logto.app}`);
+    const proxyUrl = new URL(`http://localhost:${port}`);
+
+    const proxyLogtoRequest = createProxy(
+      logtoEndpointUrl.href,
+      async (proxyResponse, request, response) =>
+        createLogtoResponseHandler({
+          proxyResponse,
+          request,
+          response,
+          logtoEndpointUrl,
+          proxyUrl,
+        })
+    );
+    const proxySignInExpRequest = createProxy(expUri);
+
+    const server = http.createServer((request, response) => {
+      consoleLog.info(`Incoming request: ${chalk.blue(request.url)}`);
```

Comment:
1. this should be an optional behavior. e.g. vite will send numerous requests in dev mode.
2. can be prettier, e.g. `GET /some/url`

Reply:
OK, then we can add a `--verbose` arg?